### PR TITLE
Run pub with --trace

### PR DESCRIPTION
**Pull request by Hixie**
See https://github.com/dart-lang/pub/issues/1714

### CODE REVIEW

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 1]
```diff
@@ -81,12 +81,18 @@ typedef String MessageFilter(String message);
...
+/// The `--trace` argument is passed to `pub` (by mutating the provided
+/// `arguments` list) unless `showTraceForErrors` is false.
 Future<Null> pub(List<String> arguments, {
   String directory,
   MessageFilter filter,
   String failureMessage: 'pub failed',
   @required bool retry,
+  bool showTraceForErrors: true,
 }) async {
+  if (!showTraceForErrors)
```
Comment (goderbauer) [inline packages/flutter_tools/lib/src/dart/pub.dart]: Should the `!` be removed to be in line with doc comment and parameter name?

Comment (Hixie) [reply packages/flutter_tools/lib/src/dart/pub.dart]: d'oh.

Comment (Hixie) [reply packages/flutter_tools/lib/src/dart/pub.dart]: fixed PTAL

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
Looks like a test is failing, though.

**Comment by Hixie** [comment]
Gotta love that we actually test this.

**Comment by Hixie** [comment]
Fixed the test. will land on green.

**Pull request merged**